### PR TITLE
Stop forcing black onto agent terminal content; add theme override toggle (Fixes #179)

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Fixes #179 — the follow-up to the theme picker work (#136 / PR #174). This addresses both the **bug** (jefe forces solid-black onto the embedded agent CLI's terminal cells) and the **enhancement** (a global opt-in to apply jefe's theme to agent panes).

## Bug: split-appearance terminal pane (requirement 1)

Set jefe's theme to GitHub Dark and the terminal pane showed a split appearance: stale/unrefreshed regions rendered with jefe's theme background, while recently-typed cells and the agent's input box rendered solid black. Typing created new written cells that flipped to black while old regions stayed themed — inconsistent within a single pane.

### Root cause (two cooperating bugs)

- `src/runtime/attach.rs` resolved terminal-default background (`Named(Background)`) to `fallback_ansi_color(0)` (solid black) instead of terminal-default.
- `src/ui/components/terminal_view.rs` unconditionally painted every run's background, overwriting the container fill with that black.

### Fix

Plain terminal-default cells now resolve to `Color::Reset`. `paint_terminal_cells` skips `set_background_color` for `Reset` backgrounds so the agent's own default/transparent cells pass through and the pane is visually consistent. Explicitly-styled cells (the agent's intentional theming) are preserved unchanged. Transformed cells (inverse, selection, cursor) retain concrete ANSI high-contrast colors so inversion and the cursor block stay visible.

A pure `resolve_run_colors` helper was extracted from the paint path so the off/on resolution matrix is unit-testable without canvas coupling.

## Enhancement: "Apply theme to agent" toggle (requirement 2)

A global checkbox (default off) in the theme picker dialog. When enabled, jefe's theme fg/bg is applied to the agent's default (`Reset`) cells while explicit cells pass through — an opt-in escape hatch to make agent panes match jefe's look.

- The toggle lives in `ModalState::ThemePicker` and is initialized from `AppState.override_agent_theme` when the picker opens.
- It is committed to `AppState` on **Enter** via the deterministic reducer (`PickerConfirm`), and discarded on **Esc** (no live mutation of `AppState` while toggling — Tab is a pure modal-flag toggle). This keeps all state transitions in the reducer (per the architecture boundary: state/event layer owns transitions) and avoids disk I/O on cancel.
- It persists to `settings.toml` as `override_agent_theme` with `#[serde(default)]` (legacy files without the field deserialize to `false`; no schema bump) and is restored at startup.

jefe's chrome (borders, title bars, sidebars, status bar, modals, lists) remains themed in all cases; only embedded terminal content behavior changes.

## Tests

- `src/runtime/attach_tests.rs`: default-cell transparency, explicit Spec/Indexed preservation, mixed default-bg/explicit-fg, base-style Reset, and inverse/cursor cells keeping concrete contrast.
- `src/ui/components/terminal_view.rs`: `resolve_run_colors` off/on matrix (default-bg transparent, explicit passthrough, override maps defaults, override leaves explicit, mixed).
- `tests/ui/theme_picker_behavior.rs`: toggle flips modal flag, double-toggle restores, noop when picker closed, open initializes from AppState, confirm commits to AppState, cancel does not commit.
- `src/persistence/tests.rs`: legacy TOML without the field deserializes to `false`; true survives a save/load round-trip.
- `dev-docs/tmux-scenarios/theme-override-toggle.json`: TUI scenario (F9 → Tab → Enter).

## Verification

- `cargo fmt --all --check` — clean
- `cargo clippy --workspace --all-targets --all-features --locked -- -D warnings` — clean
- `make ci-check` (fmt, clippy-allow gate, source-file-size gate, clippy+complexity, coverage >= 30%, build, test) — passes; 257 tests + 1 doctest green

Closes #179.

### CODE REVIEW

File: src/ui/components/terminal_view.rs [hunk 1]
```diff
@@ -223,12 +244,86 @@ impl Component for TerminalGrid {
...
+}
+
 /// Paint the styled terminal cells onto the canvas as style-runs.
 fn paint_terminal_cells(
     canvas: &mut CanvasSubviewMut<'_>,
     snapshot: &TerminalSnapshot,
     max_rows: usize,
     max_cols: usize,
+    theme_override: TerminalThemeOverride,
 ) {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/terminal_view.rs:416]: Addressed in 951b95e. `resolve_run_colors` now normalizes a `Reset` theme channel to a concrete fallback (black bg / white fg) when override is enabled, so override always paints an opaque, visible result regardless of how the override colors are sourced. `ResolvedColors` always supplies concrete `Rgb` today (parse_hex returns `Rgb` with an `Rgb` fallback), so this is a defensive contract guarantee for the pure function. Added two tests (resolve_override_normalizes_reset_theme_bg_to_opaque, resolve_override_normalizes_reset_theme_fg_to_visible) covering the edge case for both channels.

File: tests/ui/theme_picker_behavior.rs [hunk 2]
```diff
@@ -165,3 +167,97 @@ fn view_marks_active_theme_independently_of_selection() {
     assert!(!rows[1].selected, "row 1 should not be selected");
     assert!(rows[1].active, "row 1 should be active");
 }
+
+// ============================================================================
+// Override theme toggle (issue #179)
+// ============================================================================
+
+#[test]
+fn toggle_override_flips_modal_flag() {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply tests/ui/theme_picker_behavior.rs]: Addressed in 800d6f8: renamed to `toggle_override_flips_override_flag` so the name describes what is verified (the override toggle) rather than where it is stored.

File: tests/ui/theme_picker_behavior.rs [hunk 3]
```diff
@@ -165,3 +167,97 @@ fn view_marks_active_theme_independently_of_selection() {
...
+
+#[test]
+fn toggle_override_is_noop_when_picker_not_open() {
+    let state = AppState::default().apply(AppEvent::ThemePickerToggleOverride);
+    assert_eq!(
+        state.modal,
+        ModalState::None,
+        "toggle should be a no-op when picker is not open"
+    );
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply tests/ui/theme_picker_behavior.rs:234]: Addressed in 800d6f8: added `assert!(!state.override_agent_theme, ...)` so the test now verifies the app-level override flag is unchanged when the toggle fires while the picker is closed.

File: src/runtime/attach.rs [hunk 4]
```diff
@@ -338,31 +341,83 @@ fn snapshot_position(indexed: &Indexed<&Cell>, rows: usize, cols: usize) -> Opti
...
+    };
 
     TerminalCellStyle {
-        fg: rgb_to_iocraft(fg),
-        bg: rgb_to_iocraft(bg),
+        fg: to_terminal(fg_is_default, channels[0]),
+        bg: to_terminal(bg_is_default, channels[1]),
         bold: indexed.cell.flags.intersects(Flags::BOLD | Flags::DIM_BOLD),
         underline: indexed.cell.flags.intersects(Flags::ALL_UNDERLINES),
     }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/runtime/attach.rs:437]: Addressed in 72ed4d5. Added a `dim: bool` field to `TerminalCellStyle` so the DIM flag survives separately from the transparent `Color::Reset` foreground. The renderer maps it to `Weight::Light` (ANSI SGR 2). The flag is set only for plain default-colored cells, so explicitly-colored DIM cells (whose intensity is already baked into the RGB via `dim_rgb`) are not double-dimmed. Covered by `dim_default_cell_keeps_transparent_fg_but_sets_dim_flag`, `dim_bold_default_cell_sets_both_bold_and_dim_flags`, and `dim_explicit_cell_bakes_dimming_into_color_not_flag`.

File: src/runtime/attach.rs [hunk 5]
```diff
@@ -338,31 +341,83 @@ fn snapshot_position(indexed: &Indexed<&Cell>, rows: usize, cols: usize) -> Opti
...
+        channels[1] = fallback_ansi_color(7);
+    }
+    let is_cursor = cursor.shape != ansi::CursorShape::Hidden && indexed.point == cursor.point;
+    if is_cursor {
+        channels.swap(0, 1);
+    }
+
+    // After all transforms: a plain default channel maps to Reset (transparent)
+    // so the host terminal's colors show through; everything else is concrete.
+    let plain = !indexed.cell.flags.contains(Flags::INVERSE) && !is_selected && !is_cursor;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/runtime/attach.rs:410]: Addressed in 72ed4d5 via the separate `dim` field approach (rather than excluding DIM from the `plain` condition). The dimmed RGB is still discarded for default-colored cells (transparent fg), but the DIM flag is now carried on `TerminalCellStyle.dim` and rendered as `Weight::Light`, preserving the dimming without losing transparency. The alternate suggestion of preserving the concrete dimmed color would have broken the transparency contract for default-colored text.

File: src/ui/components/terminal_view.rs [hunk 6]
```diff
@@ -508,4 +675,184 @@ mod tests {
...
+            terminal_content_background(false, Color::Blue),
+            Color::Reset
+        );
+    }
+
+    #[test]
+    fn content_background_is_theme_bg_when_override_on() {
+        // Override ON: content area is filled with jefe's theme bg.
+        assert_eq!(terminal_content_background(true, Color::Blue), Color::Blue);
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui/components/terminal_view.rs:710]: Addressed in 72ed4d5. Added `content_background_passes_reset_theme_bg_through_when_override_on`, which documents that the container fill deliberately forwards a Reset theme bg unchanged (opaqueness for default cells is handled per-cell by `resolve_run_colors`, not by the container fill).

File: src/persistence/tests.rs [hunk 7]
```diff
@@ -141,6 +142,60 @@ fn file_persistence_roundtrip_settings() {
...
+
+    mgr.save_settings(&settings).value_or_panic("should save");
+    let loaded = mgr.load_settings().value_or_panic("should load");
+
+    assert!(
+        loaded.override_agent_theme,
+        "override_agent_theme must survive a save/load round-trip when true"
+    );
+
+    let _ = std::fs::remove_dir_all(&temp);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/persistence/tests.rs:196]: Acknowledged, but leaving as-is. Manual `remove_dir_all` is the established convention across every test in this file (dozens of tests use the same pattern). Tests run in isolated processes, so a leaked temp dir from a panicking test does not affect other tests. Switching this one test to the `tempfile` crate while leaving the rest inconsistent would be churn without consistent benefit; a repo-wide migration would be a separate, out-of-scope change.

File: tests/ui/theme_picker_behavior.rs [hunk 8]
```diff
@@ -165,3 +167,101 @@ fn view_marks_active_theme_independently_of_selection() {
...
+
+// ============================================================================
+// Override theme toggle (issue #179)
+// ============================================================================
+
+#[test]
+fn toggle_override_flips_override_flag() {
+    let state = picker_state(&["a", "b"], 0).apply(AppEvent::ThemePickerToggleOverride);
+    let override_val = theme_picker_override_view(&state)
+        .unwrap_or_else(|| panic!("picker should be open with override field"));
```
[the remark this answers is not in the document]
Comment (acoliver) [reply tests/ui/theme_picker_behavior.rs:195]: Leaving as-is. The `unwrap_or_else(|| panic!(...))` form with a descriptive message is the established convention in this test file (used 9 times; `expect` is not used here). The message text is already descriptive, and converting only these call sites would be inconsistent with the rest of the file.

File: tests/ui/theme_picker_behavior.rs [hunk 9]
```diff
@@ -165,3 +167,101 @@ fn view_marks_active_theme_independently_of_selection() {
...
+        theme_picker_override_view(&state).unwrap_or_else(|| panic!("picker should still be open"));
+    assert!(
+        !override_val,
+        "double toggle should return override_theme to false"
+    );
+}
+
+#[test]
+fn toggle_override_is_noop_when_picker_not_open() {
+    let state = AppState::default().apply(AppEvent::ThemePickerToggleOverride);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply tests/ui/theme_picker_behavior.rs]: Addressed in 53a5442. The test now constructs the initial state with explicit `modal: ModalState::None` and `override_agent_theme: false` preconditions instead of relying on `AppState::default()`.

File: tests/ui/theme_picker_behavior.rs [hunk 10]
```diff
@@ -165,3 +167,101 @@ fn view_marks_active_theme_independently_of_selection() {
...
+    // Cancel discards the in-dialog toggle; the runtime mirror is unchanged.
+    let state = picker_state(&["a", "b"], 0)
+        .apply(AppEvent::ThemePickerToggleOverride)
+        .apply(AppEvent::CloseThemePicker);
+    assert_eq!(state.modal, ModalState::None);
+    assert!(
+        !state.override_agent_theme,
+        "cancel must not commit the in-dialog override toggle"
+    );
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply tests/ui/theme_picker_behavior.rs:303]: Addressed in 53a5442. Added `confirm_commits_override_false_to_app_state`, which starts with the in-dialog override ON, toggles it OFF, confirms, and asserts `override_agent_theme` is false. The commit logic is now covered in both directions.

File: dev-docs/tmux-scenarios/theme-override-toggle.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    "initial_wait_ms": 100,
+    "assert_mode": "strict"
+  },
+  "steps": [
+    { "waitFor": "LLxprt Jefe" },
+    { "key": "F9" },
+    { "waitFor": "Select Theme" },
+    { "expect": "Apply theme to agent" },
+    { "key": "Tab" },
+    { "expect": "[x]" },
```
[the remark this answers is not in the document]
Comment (acoliver) [reply dev-docs/tmux-scenarios/theme-override-toggle.json:16]: Good catch. Added the baseline `{ "expect": "[ ]" }` assertion before the Tab keypress so a regression where Tab is a no-op (checkbox already checked) would be caught. Including in the next push.
